crates/astria-sequencer/src/mempool.rs

```diff
@@ -117,6 +130,98 @@ impl std::hash::Hash for EnqueuedTransaction {
     }
 }
 
+/// inserts or updates the transaction in a timestamp preserving manner
+fn update_or_insert(
```

Comment:
@Fraser999 is this an okay place to put a function like this? The rust CLI suggested I move it out of the Mempool struct's implementation because it wasn't using the `self` variable

Reply:
Absolutely.  Another choice would be to leave it inside the `impl Mempool` but just don't have `self` as a parameter.  You have to call it like `Self::update_or_insert` in that case.  It can make sense to do that in cases like this where you have more than struct in scope which _could_ use this function, but it's really only intended to be used by the `Mempool` struct.  But for a private function like this, it's pretty much down to personal preference :)

You _could_ simplify the signature though by just taking `queue: &mut PriorityQueue` and at callsites passing `&mut *self.queue.write().await`.  Definitely not a big deal though :)